pennylane/transforms/metric_tensor.py

```diff
@@ -252,14 +261,25 @@ def circuit(weights):
         )
         return [], lambda _: ()
 
+    if argnum is None:
+        argnum = tape.trainable_params
+    if isinstance(argnum, int):
+        argnum = [argnum]
+    if any(i not in tape.trainable_params for i in argnum):
+        warnings.warn(
+            "Some parameters specified in argnum are not in the "
+            f"trainable parameters {tape.trainable_params} of the tape "
+            "and will be ignored."
+        )

```

Comment:
This is difficult for me to judge. Since it is not a critical problem, I thought it might be more user friendly to only output a warning here. But I think both solutions are reasonable.

Reply:
Sounds good. Being consistent is definitely the right choice, I think.